src/actions/currency.ts

```diff
@@ -40,13 +40,11 @@ export function CurrencyActions<ActionsBase extends Constructor<TinlakeParams>>(
     }
 
     approveSeniorForCurrency = async (currencyAmount: string) => {
-      if (!this.contractAddresses['SENIOR_TRANCHE']) return
-      return this.approveCurrency(this.contractAddresses['SENIOR_TRANCHE'], currencyAmount)
+      return this.approveCurrency(this.contract('SENIOR_TRANCHE').address, currencyAmount)
```

Comment:
When dealing with async functions it is best to throw and Error. I am assuming that the check was there for a reason and It might happen not to have that set. Return undefined is very poorly handled and I agree that a promise should not return undefined, it should resolve or reject.

Maybe add something like:
```
if (!this.contractAddresses['SENIOR_TRANCHE'])
 throw new Error('Senior tranche address is undefined')
```